src/components/TodoInfo/TodoInfo.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useEffect, useState } from 'react';
+import cn from 'classnames';
+import { TempTodo, Todo } from '../../types/Todo';
+import { removeTodo, updateTodo } from '../../api/todos';
+import { ErrorType } from '../../types/ErrorType';
+
+type Props = {
+  todo: Todo | TempTodo,
+  todos?: Todo[],
+  setTodos?: (todos: Todo[]) => void,
+  setError?: (message: ErrorType) => void,
+  addedTodoIsLoading?: boolean,
+};
+export const TodoInfo: React.FC<Props> = ({
+  todo,
+  setTodos,
+  todos,
+  setError,
+  addedTodoIsLoading,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [newTitle, setNewTitle] = useState(todo.title);
+
+  const deleteTodo = async () => {
+    setLoading(true);
+    const filteredTodos = todos?.filter(({ id }) => id !== todo.id);
+
+    try {
+      await removeTodo(todo.id);
+      if (setTodos) {
```

Comment:
what means `setTodos`? Are you checking whether there is a function?

Reply:
Because for types i made it optional  so  i need that checking.
what is your suggestion ? @naziamoff